helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -412,6 +430,61 @@ private void refreshBaseline(ResourceControllerDataProvider clusterData,
     return newAssignment;
   }
 
+  private void asyncReportBaselineDivergenceGauge(ResourceControllerDataProvider clusterData,
+      Map<String, ResourceAssignment> baseline,
+      Map<String, ResourceAssignment> bestPossibleAssignment) {
+    asyncExecute(clusterData.getAsyncTasksThreadPool(), () -> {
+      try {
+        reportBaselineDivergenceGauge(baseline, bestPossibleAssignment);
+      } catch (Exception e) {
+        LOG.error("Failed to report BaselineDivergenceGauge metric.", e);
+      }
+      return null;
+    });
+  }
+
+  private void reportBaselineDivergenceGauge(Map<String, ResourceAssignment> baseline,
+      Map<String, ResourceAssignment> bestPossibleAssignment) {
+    double baselineDivergence = measureBaselineDivergence(baseline, bestPossibleAssignment);
+
+    BaselineDivergenceGauge baselineDivergenceGauge = _metricCollector.getMetric(
+        WagedRebalancerMetricCollector.WagedRebalancerMetricNames.BaselineDivergenceGauge.name(),
+        BaselineDivergenceGauge.class);
+
+    baselineDivergenceGauge.updateValue(baselineDivergence);
+  }
```

Comment:
Either make this one method to prevent misuse (because `reportBaselineDivergenceGauge` is to be called in an async manner only) or rename it?

This makes it looks like you could either report it async or sync. We just want the former, correct?